packages/scaffold-ui/src/partials/w3m-connect-recent-widget/index.ts

```diff
@@ -42,13 +44,29 @@ export class W3mConnectRecentWidget extends LitElement {
 
   // -- Render -------------------------------------------- //
   public override render() {
+    const currentNamespace = ChainController.state.activeChain
     const recentWallets = StorageUtil.getRecentWallets()
-    const filteredRecentWallets = recentWallets.filter(
-      wallet =>
-        !this.connectors.some(
-          connector => connector.id === wallet.id || connector.name === wallet.name
-        )
-    )
+
+    const filteredRecentWallets = recentWallets
+      .filter(
+        wallet =>
+          !this.connectors.some(
+            connector => connector.id === wallet.id || connector.name === wallet.name
+          )
+      )
+      .filter(wallet => {
+        if (currentNamespace && wallet.chains) {
+          const hasMatchingNamespace = wallet.chains.some(c => {
+            const chainNamespace = c.split(':')[0] as ChainNamespace
+
+            return currentNamespace === chainNamespace
+          })
+
+          return hasMatchingNamespace
+        }
+
+        return true
+      })
```

Comment:
What about doing a one iteration? That'd be more performant. Maybe we could move the logic to helper util like `filterRecentWallets` and pass the function to filter, like :
```suggestion
      .filter(filterRecentWallets)
```